### DISCUSSION TITLE
Using the Perception module

### DISCUSSION THREAD

**Discussion by julliomendonca** [eclipse/mosaic#219] [Q&A]
Hello, 

I'm trying to use the new Perception module implemented on MOSAIC 22. 
I'm running just the example app (org.eclipse.mosaic.app.tutorial.vehicle.PerceptionApp), but I keep receiving empty values for the perceived vehicles in the log, e.g.,( `- Perceived vehicles: []`)
I checked the tools version I'm using (MOSAIC 22, SUMO 1.13.0, Java JDK 8) and the dependencies libraries, and everything seems ok. 
Also, the scenario I'm using has many vehicles in it.  

Was anyone able to run this module?
Does the perception module need anything else to run?

**Comment by schwepmo** [comment]
It actually should somewhat work out of the box. I'm assuming you've made sure that there actually should be vehicles in viewing range and the Perception Module is properly enabled.
Now one thing that comes to mind, is that the Perception Module requires all vehicles to be subscribed in SUMO. And since you said that you are dealing with a large scenario, you might have disabled that option. (See here https://www.eclipse.org/mosaic/docs/mosaic_configuration/sumo_config/ under `subscribeToAllVehicles`)

Also some documentation on the Perception Module is coming to our website soon.

**Reply by julliomendonca** [reply]
Hi, thanks for the reply. 

I managed to use the Perception Module. It seems that there is some issue with my sumo files and/or the database. I was using a sumo net file created by myself. 
The Perception module worked well with the sumo and databases files from the examples scenarios and from an exported road that I created using the scenario convert tool. 
However, could not figure out yet what is the problem with my personalized sumo net or app db file.

**Reply by schwepmo** [reply]
Indeed this could be an issue as the Perception Module relies on the projection being set up properly, which usually isn't the case when you create a scenario with netedit. You can try using `--import-zone 32n` as additional parameter for scenario convert, which is what we did for our test scenario. However your coordinates will not correlate to any real world locations.
